### ISSUE TITLE
Add option to use password-protected accounts on the same Daedalus instance

### ISSUE THREAD

**Issue by vantuz-subhuman** [input-output-hk/daedalus#853]
For now, Daedalus has completely open and transparent interface, that is - anyone who opens it, may see all the data and all the wallets, and only restricted from spending money from a wallet if it does have a spending password.

1) This assumes a single computer or a computer account being used by a single person. While this assumption probably holds for the majority - there are still a lot of people who often share the same computer or even the same account.

2) If someones computer is stolen - having a "spending password" will protect them from getting their assets spent, but it will in no way protect their personal information from being openly observable.

Both of these points may be critical for some users, who want to use the wallet software but in the same time is not willing to expose the personal information of their wallets.

It is additionally discussed in this thread: https://forum.cardano.org/t/opening-the-wallet-without-a-password/9136

It would be very nice to have these options:

1) By default, Daedalus is open in the same way as now - but there's an option in settings to set a password to it. When password set - there's still a single account on that Daedalus, but when it's launched - user is prompted to: either enter a password, or create new empty account.

2) Whenever Daedalus is launched - there's always an additional option in settings, to "create new account". When new account is created with a name (default one may be called "default" or "main" automatically with an option to rename it) - at the next start Daedalus prompts user to select one of the accounts to be used. When user have selected one of the accounts (if there're multiple of them) - there's an additional option to "log out" that would move user back to the account-selection screen. Each account may be optionally protected by a password on its settings page.

Multiple accounts in the same Daedalus instance seem to work nicely together, since all of them ma share the same chain-data. So whenever the Daedalus is launched - it first downloads the missing chain-tail and then proceeds to work with any of the selected accounts.

Basically (in simplified view, of course) it would only require to keep a mapping of "account authentication data" to the "Wallet DB" with the seed, address caches, etc. So basically like a multiple wallet collections on top of the same chain, identifiable by an "account name" and protected by a password.

Of course, spending password should still be available for any of users and wallets.

**Comment by CaptJakk** [comment]
We 100% need to create an option to password protect getting into the wallet at all. The fact that someone can just click the application and find out how much is stored in it without having to type in the password is not really tenable in the long run. Ideally the keys should be encrypted on disk and only be initially decrypted when that password was typed in.

**Comment by smenroll** [comment]
Very good points @vantuz-subhuman! 
And something that is not related to a multi user option, but more related to spending password:
Perhaps we should recommend setting a spending password to *ON* as the default. Today when you create a wallet the default is not having a spending password, and that is a bit insecure if novice users forget to set a spending password. At least have a reminder that we recommend setting a spending password. Today as of "Version 1.1.5813" the default is no spending password.

**Comment by SebastienGllmt** [comment]
@smenroll this is already included in the 0.10.0 release: https://github.com/input-output-hk/daedalus/pull/856

**Comment by smenroll** [comment]
@SebastienGllmt ok, good catch! 

Ok, then, on the "first wallet being created" in Deadalus, user is defaulted to turning spending password on", but when I try to create a new additional wallet, "the spending password is by default off", for this new second wallet (just thought I would point that out, as it might make sense to have it on all new wallets being create also (not just the first one being created)?
 
![create new deadalus wallet 2](https://user-images.githubusercontent.com/39009659/39722681-6cad4ee0-51f8-11e8-9cc6-588c8795ff71.png)
![password not enabled by default 2](https://user-images.githubusercontent.com/39009659/39722694-744f5ddc-51f8-11e8-82f8-4a5ce0daaf5a.png)

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
